google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/SubscriptionAdminClient.java

```diff
@@ -344,7 +345,7 @@ public final Subscription createSubscription(
    * <pre><code>
    * try (SubscriptionAdminClient subscriptionAdminClient = SubscriptionAdminClient.create()) {
    *   ProjectSubscriptionName name = ProjectSubscriptionName.of("[PROJECT]", "[SUBSCRIPTION]");
-   *   TopicName topic = ProjectTopicName.of("[PROJECT]", "[TOPIC]");
+   *   TopicName topic = TopicName.ofProjectTopicName("[PROJECT]", "[TOPIC]");
```

Comment:
We will have to talk at the client meeting on Thursday about updating the many code examples that exist. I don't think it's good for us to have code examples in different places showing different things.

Reply:
So `TopicName.ofProjectTopicName()` will be the way to instantiate a resource name going forward as we are deprecating the per-pattern resource name classes. It will work as of this PR. 

Our generator is stateless so we will be unable to continue to generate legacy code sample `ProjectTopicName.of(...)` in this case. Although IMHO it isn't worth doing so as it would add maintenance cost for the future and inconsistency among cloud APIs horizontally, I can convert it back in the postprocessing step. Let me know if you want to take this approach.

Reply:
I'm fine with documentation being updated to use the new version. It's just we have sample code in a bunch of places on the Cloud website that needs to be updated, too. The more coordinated that effort can be, the better.

Reply:
Okay that makes sense. Just to be clear (there's a lot of overloaded term here), when you say "I am fine with documentation being updated", you are talking about this specific change in this PR, right? (Because we actually refer to this as samples while documentation is the pure Javadoc from the perspective of code generators)

I actually think you can't do both (merge this PR and update the samples) at the same time - you will need to merge this PR, release a new artifact to maven and then update samples to depend on it and rewrite the samples. Otherwise the samples won't compile.

Reply:
I mean I am okay with our code examples throughout the entirety of the Cloud Pub/Sub website being updated to reflect these new methods. We will have to do that over time once this is submitted and released.